cliotestutils/application.go

```diff
@@ -0,0 +1,73 @@
+package cliotestutils
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/anchore/clio"
+)
+
+// NewApplication takes a testing.T, a clio setup config, and a slice of assertions, and returns
+// a clio application that will, instead of setting up commands with their normal RunE, set up commands
+// such that the assertions are called with the testing.T after config state is set up by reading flags,
+// env vars, and config files. Useful for testing that expected configuration options are wired up.
+// Note that initializers will be cleared from the clio setup config, since the initialization may happen
+// more than once and affect global state. For necessary global state, a workaround is to set it in a TestingMain.
+func NewApplication(t *testing.T, cfg *clio.SetupConfig, assertions ...AssertionFunc) clio.Application {
```

Comment:
I like the composition approach here - it might get tough to debug this, since the assertions can happen so far away from the thing they are asserting against, but I think that's a second order concern.

Reply:
👍